### PR TITLE
synce4l on dpll v2

### PR DESCRIPTION
**Pull request by kubalewski**
Adding support for dpll based synce4l flow.
Currently supports only dpll in automatic mode, for manual mode dplls it
will not work, this shall be added in the future.
Detailed configuration changes available in:
"synce4l: add support for dpll subsystem related config items" commit.

v2:
- redesign behavior of dpll_mon/state machine for easier reinit
- remove async read of synchronous sockets
- fix dpll_rt_recv(..) behavior
- fix crash with one source configured
- netlink fixes/improvements
- reduce cpu usage
- add info logs for external source holdover/locked states
- fix function argument doxygens

### CODE REVIEW

File: config.c [hunk 1]
```diff
@@ -294,10 +313,11 @@ static enum parser_result parse_item(struct config *cfg,
 				     const char *option,
 				     const char *value)
 {
-	enum parser_result r;
 	struct config_item *cgi, *dst;
 	struct config_enum *cte;
+	enum parser_result r;
 	double df = 0.0;
+	uint64_t u64;
```
Comment (mlichvar) [inline config.c:320]: This new u64 variable triggers -Wmaybe-uninitialized warning for me. Can you please initialize it?

**Comment by mlichvar** [comment]
All issues I had with the previous PR seem to be fixed. Thanks!

In my tests it's working well. I tried it as a SyncE source with GPS and receiver, with one port and two ports. It interoperates with older synce version using the out-of-tree ice driver.

File: Makefile [hunk 2]
```diff
@@ -17,13 +17,23 @@ else
...
 
-CC	= gcc
-CFLAGS	= -Wall -Wextra -Werror $(EXTRA_CFLAGS) -pthread -DVERSION=$(VERSION)
-LDLIBS	= -lm -lrt -pthread $(EXTRA_LDFLAGS)
+LINUX_HDR_PATH   = /usr/include
+ifneq ("$(wildcard $(LINUX_HDR_PATH)/linux/dpll.h)","")
+  $(info Using linux/dpll.h header found in $(LINUX_HDR_PATH))
+  DPLL = -DCONFIG_DPLL
+else
+  $(warning linux/dpll.h header not found - using backup file: headers/dpll.h)
```
[the remark this answers is not in the document]
Comment (kubalewski) [reply Makefile:25]: Well makes sense to me, fix in v3 https://github.com/intel/synce4l/pull/26

**Comment by kubalewski** [comment]
> All issues I had with the previous PR seem to be fixed. Thanks!
> 
> In my tests it's working well. I tried it as a SyncE source with GPS and receiver, with one port and two ports. It interoperates with older synce version using the out-of-tree ice driver.

Thanks, sound great, please try the new version https://github.com/intel/synce4l/pull/26

**Comment by kubalewski** [comment]
closing as new version available

**Pull request closed without merging**